pkg/cmd/node/node_diagnostics.go

```diff
@@ -0,0 +1,55 @@
+package node
+
+import (
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdNodeMetrics creates the `core node metrics` subcommand.
+func newCmdNodeDiagnostics(ctx api.Context) *cobra.Command {
+	var cancel bool
+	var jsonOutput bool
+	var list bool
+	var status bool
+	cmd := &cobra.Command{
+		Use:   "diagnostics",
+		Short: "Use diagnostics bundles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var subCommand *cobra.Command
+			switch {
+			case cancel && !list && !status:
+				subCommand = newCmdNodeDiagnosticsCancel(ctx)
+			case !cancel && list && !status:
+				subCommand = newCmdNodeDiagnosticsList(ctx)
+			case !cancel && !list && status:
+				subCommand = newCmdNodeDiagnosticsStatus(ctx)
+			}
+
+			if subCommand != nil {
+				subCommand.SilenceErrors = true
+				subCommand.SilenceUsage = true
+				subCommand.SetArgs(ctx.Args()[4:])
+				return subCommand.Execute()
+			}
+			return cmd.Help()
+		},
+	}
+	cmd.Flags().BoolVar(&cancel, "cancel", false, "Cancel a running diagnostics job")
+	cmd.Flags().MarkDeprecated("cancel", "use the 'cancel' subcommand instead")
+	cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
```

Comment:
this can be `cmd.Flags().Bool("json", false, "Print in json format")`, the variable is not needed as it's not being used, it's just for cobra to register the option.